mythril/support/signatures.py

```diff
@@ -1,44 +1,244 @@
+    # fake it :)
+    ethereum_input_decoder = None
+    FourByteDirectoryOnlineLookupError = Exception
+
+
+class SimpleFileLock(object):
+    # todo: replace with something more reliable. this is a quick shot on concurrency and might not work in all cases
+
+    def __init__(self, path):
+        self.path = path
+        self.lockfile = pathlib.Path("%s.lck" % path)
+        self.locked = False
+
+    def aquire(self, timeout=5):
+        if self.locked:
+            raise Exception("SimpleFileLock: lock already aquired")
+
+        t_end = time.time()+timeout
+        while time.time() < t_end:
+            # try to aquire lock
+            try:
+                self.lockfile.touch(mode=0o0000, exist_ok=False)  # touch the lockfile
+                # lockfile does not exist. we have a lock now
+                self.locked = True
+                return
+            except FileExistsError as fee:
+                # check if lockfile date exceeds age and cleanup lock
+                if time.time() > self.lockfile.stat().st_mtime + 60 * 5:
+                    self.release(force=True)  # cleanup old lockfile > 5mins
+
+                time.sleep(0.5)  # busywait is evil
+                continue
+
+        raise Exception("SimpleFileLock: timeout hit. failed to aquire lock: %s"% (time.time()-self.lockfile.stat().st_mtime))
+
+    def release(self, force=False):
+        if not force and not self.locked:
+            raise Exception("SimpleFileLock: aquire lock first")
+
+        try:
+            self.lockfile.unlink()  # might throw if we force unlock and the file gets removed in the meantime. TOCTOU
+        except FileNotFoundError as fnfe:
+            logging.warning("SimpleFileLock: release(force=%s) on unavailable file. race? %r" % (force, fnfe))
+
+        self.locked = False
+
+
+
+class SignatureDb(object):
+
+    def __init__(self, enable_online_lookup=True):
+        """
+        Constr
+        :param enable_online_lookup: enable onlien signature hash lookup
+        """
+        self.signatures = {}  # signatures in-mem cache
+        self.signatures_file = None
+        self.signatures_file_lock = None
+        self.enable_online_lookup = enable_online_lookup  # enable online funcsig resolving
+        self.online_lookup_miss = set()  # temporarily track misses from onlinedb to avoid requesting the same non-existent sighash multiple times
+        self.online_directory_unavailable_until = 0  # flag the online directory as unavailable for some time
+
+    def open(self, path=None):
+        """
+        Open a function signature db from json file
+
+        :param path: specific path to signatures.json; default mythril location if not specified
+        :return: self
+        """
+        if not path:
+            #  try default locations
+            try:
+                mythril_dir = os.environ['MYTHRIL_DIR']
+            except KeyError:
+                mythril_dir = os.path.join(os.path.expanduser('~'), ".mythril")
```

Comment:
This doesn't work for all os systems, a good example of how to do this can be found in mythril.py: _init_config()

Reply:
this was basically the original code that was used in the `disassembler.py`. we should probably make the path guessing logic available in a utility module in order to avoid code duplications. Also the code in `_init_config()` does not appear to match the code in `_init_mythril_dir()`.
I would suggest to change this after merging the PR, providing a utililty function that returns the correct path and use this wherever path guessing is currently used. I wouldn't want to add this to this PR for the sake of readability.